src/librustc_typeck/check/pat.rs

```diff
@@ -1355,16 +1355,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     ) -> Ty<'tcx> {
         let err = self.tcx.types.err;
         let expected = self.structurally_resolved_type(span, expected);
-        let (inner_ty, slice_ty, expected) = match expected.kind {
+        let (element_ty, slice_ty, expected) = match expected.kind {
```

Comment:
```suggestion
        let (element_ty, slice_ty, inferred) = match expected.kind {
```
(Makes this a bit less jarring to read cause "expected" sounds like an input, but it's really an output here.)